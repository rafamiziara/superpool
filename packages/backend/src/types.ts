/**
 * Interface for a user's profile document stored in Firestore.
 */
export interface UserProfile {
  walletAddress: string
  createdAt: number
  updatedAt: number
}
