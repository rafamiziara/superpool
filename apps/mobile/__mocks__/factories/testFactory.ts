/**
 * Test Helper Factory Functions
 *
 * Centralized test utilities and helper functions for consistent testing patterns.
 * Includes render functions, test utilities, and common test data.
 */

import { render, renderHook, RenderOptions, RenderResult } from '@testing-library/react-native'
import React, { ReactElement, ReactNode } from 'react'
import { createMockRootStore } from './storeFactory'

/**
 * Common test utilities and helpers
 */

/**
 * Wait for MobX reactions to settle
 * Useful when testing computed values or reactions
 */
export const waitForMobX = async (timeout = 100): Promise<void> => {
  return new Promise((resolve) => {
    setTimeout(resolve, timeout)
  })
}

/**
 * Mock Firebase Auth user object for testing
 */
export const createMockFirebaseUser = (overrides: Record<string, unknown> = {}) => ({
  uid: 'test-user-id',
  email: 'test@example.com',
  displayName: 'Test User',
  emailVerified: true,
  photoURL: null,
  phoneNumber: null,
  ...overrides,
})

/**
 * Mock Ethereum addresses for testing
 */
export const MOCK_ETH_ADDRESS = '0x1234567890123456789012345678901234567890'
export const MOCK_ETH_ADDRESS_2 = '0x0987654321098765432109876543210987654321'

/**
 * Common test chain IDs
 */
export const TEST_CHAIN_IDS = {
  ETHEREUM_MAINNET: 1,
  POLYGON_MAINNET: 137,
  POLYGON_AMOY: 80002,
  LOCALHOST: 31337,
} as const

/**
 * Test Store Provider Component
 *
 * Provides a mock MobX store context for testing components
 */
interface TestStoreProviderProps {
  children: ReactNode
  store?: ReturnType<typeof createMockRootStore>
}

export const TestStoreProvider: React.FC<TestStoreProviderProps> = ({ children, store = createMockRootStore() }) => {
  // Create a simple context provider for testing
  // This is a simplified version that just passes the store as a prop
  // In real implementation, this would use the actual StoreContext
  const StoreContextProvider = ({ children: contextChildren }: { children: ReactNode }) => {
    // For testing purposes, we'll inject the store via React context
    // This assumes the actual StoreContext is available
    try {
      const { StoreContext } = require('../../src/stores/StoreContext')
      return React.createElement(StoreContext.Provider, { value: store }, contextChildren)
    } catch {
      // Fallback for when StoreContext is not available
      return React.createElement('div', { 'data-testid': 'mock-store-provider' }, contextChildren)
    }
  }

  return React.createElement(StoreContextProvider, null, children)
}

/**
 * Custom render options that includes store configuration
 */
interface CustomRenderOptions extends Omit<RenderOptions, 'wrapper'> {
  store?: ReturnType<typeof createMockRootStore>
  wrapper?: React.ComponentType<{ children: React.ReactNode }>
}

/**
 * Custom render function that wraps components with TestStoreProvider
 * This is the recommended way to render components that use MobX stores in tests
 */
export const renderWithStore = (ui: ReactElement, options: CustomRenderOptions = {}): RenderResult => {
  const { store = createMockRootStore(), wrapper: CustomWrapper, ...renderOptions } = options

  const Wrapper: React.FC<{ children: React.ReactNode }> = ({ children }) => {
    if (CustomWrapper) {
      return React.createElement(TestStoreProvider, { store, children: React.createElement(CustomWrapper, { children }) })
    }

    return React.createElement(TestStoreProvider, { store, children })
  }

  return render(ui, { wrapper: Wrapper, ...renderOptions })
}

/**
 * Custom renderHook function that wraps hooks with TestStoreProvider
 * This is the recommended way to test hooks that use MobX stores
 */
export const renderHookWithStore = <TResult>(hook: () => TResult, options: { store?: ReturnType<typeof createMockRootStore> } = {}) => {
  const { store = createMockRootStore() } = options

  const Wrapper: React.FC<{ children: React.ReactNode }> = ({ children }) => {
    return React.createElement(TestStoreProvider, { store, children })
  }

  return renderHook(() => hook(), { wrapper: Wrapper })
}

/**
 * Higher-order component that wraps a component with TestStoreProvider
 */
export const withMockStore = <P extends object>(Component: React.ComponentType<P>, store?: ReturnType<typeof createMockRootStore>) => {
  const WrappedComponent: React.FC<P> = (props) =>
    React.createElement(TestStoreProvider, { store, children: React.createElement(Component, props) })

  WrappedComponent.displayName = `withMockStore(${Component.displayName || Component.name})`

  return WrappedComponent
}

/**
 * Utility function to create a minimal render for non-store components
 * Use this only when you need to test components that don't use MobX stores
 */
export const renderWithoutStore = render

/**
 * Mock Store Presets for common test scenarios
 */
export const mockStorePresets = {
  // Connected wallet with authenticated user
  authenticatedWithWallet: () =>
    createMockRootStore({
      authenticationStore: {
        authError: null,
        authLock: {
          isLocked: false,
          startTime: 0,
          walletAddress: null,
          abortController: null,
          requestId: null,
        },
      },
      walletStore: {
        isConnected: true,
        address: MOCK_ETH_ADDRESS,
        chainId: 137,
        connectionError: null,
        isConnecting: false,
      },
    }),

  // Unauthenticated user
  unauthenticated: () =>
    createMockRootStore({
      walletStore: {
        isConnected: false,
        address: undefined,
        chainId: undefined,
        connectionError: null,
        isConnecting: false,
      },
    }),

  // Authentication in progress
  authenticating: () =>
    createMockRootStore({
      authenticationStore: {
        authLock: {
          isLocked: true,
          startTime: Date.now(),
          walletAddress: MOCK_ETH_ADDRESS,
          abortController: new AbortController(),
          requestId: 'test-request-id',
        },
      },
    }),

  // Wallet connection in progress
  connectingWallet: () =>
    createMockRootStore({
      walletStore: {
        isConnected: false,
        address: undefined,
        chainId: undefined,
        connectionError: null,
        isConnecting: true,
      },
    }),

  // Error states
  authenticationError: () =>
    createMockRootStore({
      authenticationStore: {
        authError: {
          name: 'AppError',
          message: 'Authentication failed',
          type: 'AUTHENTICATION_FAILED',
          userFriendlyMessage: 'Authentication failed',
          timestamp: new Date(),
        },
      },
    }),

  walletConnectionError: () =>
    createMockRootStore({
      walletStore: {
        isConnected: false,
        address: undefined,
        chainId: undefined,
        connectionError: 'Failed to connect wallet',
        isConnecting: false,
      },
    }),

  // Pool management with data
  poolWithData: () => {
    const mockPool = {
      id: 'test-pool-1',
      name: 'Test Pool',
      description: 'A test pool for unit tests',
      contractAddress: MOCK_ETH_ADDRESS,
      creator: MOCK_ETH_ADDRESS,
      admins: [MOCK_ETH_ADDRESS],
      members: [MOCK_ETH_ADDRESS],
      maxMembers: 10,
      minimumContribution: BigInt(50),
      interestRate: 500,
      loanDuration: 2592000,
      totalLiquidity: BigInt(1000),
      availableLiquidity: BigInt(800),
      totalBorrowed: BigInt(200),
      isActive: true,
      isPaused: false,
      createdAt: new Date('2023-01-01'),
      updatedAt: new Date('2023-01-02'),
    }

    const store = createMockRootStore({
      walletStore: {
        isConnected: true,
        address: MOCK_ETH_ADDRESS,
        chainId: 137,
      },
      poolManagementStore: {
        userAddress: MOCK_ETH_ADDRESS,
      },
    })

    // Add test pool data
    store.poolManagementStore.addPool(mockPool)

    return store
  },

  // Loading states
  loadingPools: () =>
    createMockRootStore({
      poolManagementStore: {
        loading: {
          pools: true,
          loans: false,
          transactions: false,
          memberActions: false,
        },
      },
    }),

  loadingMemberActions: () =>
    createMockRootStore({
      poolManagementStore: {
        loading: {
          pools: false,
          loans: false,
          transactions: false,
          memberActions: true,
        },
      },
    }),
}

/**
 * Re-export store factory functions for convenience
 */
export { createMockRootStore } from './storeFactory'

/**
 * Re-export common testing library functions for convenience
 */
export * from '@testing-library/react-native'

// Override the default render to use renderWithStore by default
export { renderWithStore as render }
