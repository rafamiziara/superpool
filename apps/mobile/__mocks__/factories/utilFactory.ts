/**
 * Utility Mock Factory Functions
 *
 * Reusable factory functions for creating mock utility instances with custom configurations
 */

// Toast Factory
export const createMockToast = (
  overrides: Partial<{
    showSuccess: jest.Mock
    showError: jest.Mock
    showInfo: jest.Mock
    showWarning: jest.Mock
  }> = {}
) => ({
  showSuccess: jest.fn(),
  showError: jest.fn(),
  showInfo: jest.fn(),
  showWarning: jest.fn(),
  authToasts: {
    walletConnected: jest.fn(),
    authSuccess: jest.fn(),
    walletDisconnected: jest.fn(),
    connectionFailed: jest.fn(),
    signatureRejected: jest.fn(),
    networkMismatch: jest.fn(),
    sessionRecovery: jest.fn(),
    sessionExpired: jest.fn(),
  },
  appToasts: {
    operationSuccess: jest.fn(),
    operationFailed: jest.fn(),
    loading: jest.fn(),
    dataSaved: jest.fn(),
    dataLoaded: jest.fn(),
    validationError: jest.fn(),
    permissionDenied: jest.fn(),
    offline: jest.fn(),
    online: jest.fn(),
  },
  ...overrides,
})

// Firebase Auth Manager Factory
export const createMockFirebaseAuthManager = (
  overrides: Partial<{
    getCurrentState: jest.Mock
    addListener: jest.Mock
    isAuthenticated: boolean
    user: unknown
  }> = {}
) => ({
  getCurrentState: jest.fn(() => ({
    user: null,
    isLoading: false,
    isAuthenticated: false,
    walletAddress: null,
    ...overrides,
  })),
  addListener: jest.fn((callback) => {
    callback({
      user: null,
      isLoading: false,
      isAuthenticated: false,
      walletAddress: null,
      ...overrides,
    })
    return jest.fn() // cleanup function
  }),
})

// Storage Factory (AsyncStorage/SecureStore)
export const createMockStorage = (
  overrides: Partial<{
    getItem: jest.Mock
    setItem: jest.Mock
    removeItem: jest.Mock
    clear: jest.Mock
  }> = {}
) => ({
  getItem: jest.fn(() => Promise.resolve(null)),
  setItem: jest.fn(() => Promise.resolve()),
  removeItem: jest.fn(() => Promise.resolve()),
  clear: jest.fn(() => Promise.resolve()),
  getAllKeys: jest.fn(() => Promise.resolve([])),
  ...overrides,
})
