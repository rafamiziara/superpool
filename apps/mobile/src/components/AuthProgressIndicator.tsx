import { observer } from 'mobx-react-lite'
import React from 'react'
import { Text, View } from 'react-native'
import { useAuthenticationStore } from '../stores'
import { ProgressIndicator } from './ProgressIndicator'

interface AuthProgressIndicatorProps {
  className?: string
}

export const AuthProgressIndicator = observer(function AuthProgressIndicator({ className = '' }: AuthProgressIndicatorProps) {
  const authStore = useAuthenticationStore()

  const steps = authStore.getAllSteps()
  const currentStepIndex = steps.findIndex((step) => step.step === authStore.currentStep)

  if (steps.length === 0) {
    return null
  }

  return (
    <View className={`space-y-4 ${className}`} testID="auth-progress-indicator">
      <ProgressIndicator totalSteps={steps.length} currentStep={currentStepIndex >= 0 ? currentStepIndex : -1} />

      <View className="space-y-2">
        {authStore.currentStep && (
          <Text className="text-center text-sm font-medium text-foreground" testID="current-step-title">
            {authStore.getStepInfo(authStore.currentStep)?.title}
          </Text>
        )}

        {authStore.currentStep && (
          <Text className="text-center text-xs text-muted-foreground" testID="current-step-description">
            {authStore.getStepInfo(authStore.currentStep)?.description}
          </Text>
        )}

        {authStore.progressError && (
          <Text className="text-center text-xs text-destructive" testID="progress-error">
            {authStore.progressError}
          </Text>
        )}
      </View>

      <View className="flex-row justify-center space-x-4">
        <Text className="text-xs text-muted-foreground" testID="progress-stats">
          {authStore.completedSteps.size} of {steps.length} completed
        </Text>

        {authStore.isProgressComplete && (
          <Text className="text-xs text-green-600 font-medium" testID="progress-complete">
            âœ“ Complete
          </Text>
        )}
      </View>
    </View>
  )
})
