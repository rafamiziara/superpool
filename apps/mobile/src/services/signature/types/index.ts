/**
 * TypeScript interface definitions for signature services
 * These interfaces define contracts but don't generate runtime code
 */
export type { SignatureConfig, SignatureStrategy } from './interfaces'
