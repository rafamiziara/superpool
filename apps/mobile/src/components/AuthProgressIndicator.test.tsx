import { AuthStep } from '@superpool/types'
import { act } from '@testing-library/react-native'
import React from 'react'
import { createMockRootStore, renderWithStore, waitForMobX } from '@mocks/factories/testFactory'
import { AuthProgressIndicator } from './AuthProgressIndicator'

describe('AuthProgressIndicator', () => {
  describe('Basic Rendering', () => {
    it('should render with default store state', async () => {
      const { getByTestId } = renderWithStore(<AuthProgressIndicator />)

      const container = getByTestId('auth-progress-indicator')
      expect(container).toBeTruthy()

      const progressStats = getByTestId('progress-stats')
      expect(progressStats).toBeTruthy()
    })

    it('should render null when no steps are available', async () => {
      const mockStore = createMockRootStore({
        auth: {
          getAllSteps: jest.fn().mockReturnValue([]),
          getCurrentStepInfo: jest.fn().mockReturnValue(null),
          isProgressComplete: false,
          progressError: null,
          getCompletedStepsCount: jest.fn().mockReturnValue(0),
        },
      })

      const { queryByTestId } = renderWithStore(<AuthProgressIndicator />, { store: mockStore })

      const container = queryByTestId('auth-progress-indicator')
      expect(container).toBeNull()
    })
  })

  describe('MobX Store Integration', () => {
    it('should display current step information', async () => {
      const mockStore = createMockRootStore({
        auth: {
          getAllSteps: jest.fn().mockReturnValue(['connect-wallet', 'generate-message', 'request-signature']),
          getCurrentStepInfo: jest.fn().mockReturnValue({
            title: 'Generate Auth Message',
            description: 'Creating authentication challenge',
          }),
          isProgressComplete: false,
          progressError: null,
          getCompletedStepsCount: jest.fn().mockReturnValue(1),
        },
      })

      const { getByTestId } = renderWithStore(<AuthProgressIndicator />, { store: mockStore })

      await waitForMobX()

      const currentStepTitle = getByTestId('current-step-title')
      expect(currentStepTitle.props.children).toBe('Generate Auth Message')

      const currentStepDescription = getByTestId('current-step-description')
      expect(currentStepDescription.props.children).toBe('Creating authentication challenge')
    })

    it('should not display step info when no current step', async () => {
      const mockStore = createMockRootStore({
        auth: {
          getAllSteps: jest.fn().mockReturnValue(['connect-wallet']),
          getCurrentStepInfo: jest.fn().mockReturnValue(null),
          isProgressComplete: false,
          progressError: null,
          getCompletedStepsCount: jest.fn().mockReturnValue(0),
        },
      })

      const { queryByTestId } = renderWithStore(<AuthProgressIndicator />, { store: mockStore })

      await waitForMobX()

      const currentStepTitle = queryByTestId('current-step-title')
      expect(currentStepTitle).toBeNull()

      const currentStepDescription = queryByTestId('current-step-description')
      expect(currentStepDescription).toBeNull()
    })

    it('should display progress error when available', async () => {
      const errorMessage = 'Signature verification failed'
      const mockStore = createMockRootStore({
        auth: {
          getAllSteps: jest.fn().mockReturnValue(['connect-wallet', 'verify-signature']),
          getCurrentStepInfo: jest.fn().mockReturnValue(null),
          isProgressComplete: false,
          progressError: errorMessage,
          getCompletedStepsCount: jest.fn().mockReturnValue(1),
        },
      })

      const { getByTestId } = renderWithStore(<AuthProgressIndicator />, { store: mockStore })

      await waitForMobX()

      const progressError = getByTestId('progress-error')
      expect(progressError.props.children).toBe(errorMessage)
    })

    it('should not display error when no progress error', async () => {
      const mockStore = createMockRootStore({
        auth: {
          getAllSteps: jest.fn().mockReturnValue(['connect-wallet']),
          getCurrentStepInfo: jest.fn().mockReturnValue(null),
          isProgressComplete: false,
          progressError: null,
          getCompletedStepsCount: jest.fn().mockReturnValue(0),
        },
      })

      const { queryByTestId } = renderWithStore(<AuthProgressIndicator />, { store: mockStore })

      await waitForMobX()

      const progressError = queryByTestId('progress-error')
      expect(progressError).toBeNull()
    })
  })

  describe('Reactive Updates', () => {
    it('should update when store state changes', async () => {
      let completedCount = 1
      const mockStore = createMockRootStore({
        auth: {
          getAllSteps: jest
            .fn()
            .mockReturnValue(['connect-wallet', 'generate-message', 'request-signature', 'verify-signature', 'firebase-auth']),
          getCurrentStepInfo: jest.fn().mockReturnValue(null),
          isProgressComplete: false,
          progressError: null,
          getCompletedStepsCount: jest.fn().mockImplementation(() => completedCount),
        },
      })

      const { getByTestId } = renderWithStore(<AuthProgressIndicator />, { store: mockStore })

      let progressStats = getByTestId('progress-stats')
      expect(progressStats.props.children).toEqual([1, ' of ', 5, ' completed'])

      // Update completed count and getCurrentStepInfo
      act(() => {
        completedCount = 2
        mockStore.auth.getCurrentStepInfo.mockReturnValue({
          title: 'Generate Auth Message',
          description: 'Creating authentication challenge',
        })
      })

      await waitForMobX()

      const currentStepTitle = getByTestId('current-step-title')
      expect(currentStepTitle.props.children).toBe('Generate Auth Message')

      // Complete the step
      act(() => {
        completedCount = 3
        mockStore.auth.getCurrentStepInfo.mockReturnValue(null)
      })

      await waitForMobX()

      progressStats = getByTestId('progress-stats')
      expect(progressStats.props.children).toEqual([3, ' of ', 5, ' completed'])
    })

    it('should update progress indicator when currentStep changes', async () => {
      const mockStore = createMockRootStore({
        auth: {
          getAllSteps: jest.fn().mockReturnValue(['connect-wallet', 'generate-message', 'request-signature']),
          getCurrentStepInfo: jest.fn().mockReturnValue({
            title: 'Generate Auth Message',
            description: 'Creating authentication challenge',
          }),
          isProgressComplete: false,
          progressError: null,
          getCompletedStepsCount: jest.fn().mockReturnValue(1),
        },
      })

      const { getByTestId } = renderWithStore(<AuthProgressIndicator />, { store: mockStore })

      await waitForMobX()

      let currentStepTitle = getByTestId('current-step-title')
      expect(currentStepTitle.props.children).toBe('Generate Auth Message')

      // Move to next step
      act(() => {
        mockStore.auth.getCurrentStepInfo.mockReturnValue({
          title: 'Request Signature',
          description: 'Requesting wallet signature',
        })
      })

      await waitForMobX()

      currentStepTitle = getByTestId('current-step-title')
      expect(currentStepTitle.props.children).toBe('Request Signature')
    })

    it('should show completion status when progress is complete', async () => {
      const mockStore = createMockRootStore({
        auth: {
          getAllSteps: jest.fn().mockReturnValue(['connect-wallet', 'generate-message']),
          getCurrentStepInfo: jest.fn().mockReturnValue(null),
          isProgressComplete: false,
          progressError: null,
          getCompletedStepsCount: jest.fn().mockReturnValue(1),
        },
      })

      const { getByTestId, queryByTestId } = renderWithStore(<AuthProgressIndicator />, { store: mockStore })

      await waitForMobX()

      // Initially not complete
      let progressComplete = queryByTestId('progress-complete')
      expect(progressComplete).toBeNull()

      // Complete all steps
      act(() => {
        mockStore.auth.isProgressComplete = true
        mockStore.auth.getCompletedStepsCount.mockReturnValue(2)
      })

      await waitForMobX()

      progressComplete = getByTestId('progress-complete')
      expect(progressComplete.props.children).toBe('âœ“ Complete')
    })
  })

  describe('Progress Statistics', () => {
    it('should display correct completed steps count', async () => {
      const mockStore = createMockRootStore({
        auth: {
          getAllSteps: jest.fn().mockReturnValue(['connect-wallet', 'generate-message', 'request-signature', 'verify-signature']),
          getCurrentStepInfo: jest.fn().mockReturnValue(null),
          isProgressComplete: false,
          progressError: null,
          getCompletedStepsCount: jest.fn().mockReturnValue(3),
        },
      })

      const { getByTestId } = renderWithStore(<AuthProgressIndicator />, { store: mockStore })

      await waitForMobX()

      const progressStats = getByTestId('progress-stats')
      expect(progressStats.props.children).toEqual([3, ' of ', 4, ' completed'])
    })

    it('should handle edge case with all steps completed', async () => {
      const allSteps: AuthStep[] = [
        'connect-wallet',
        'acquire-lock',
        'generate-message',
        'request-signature',
        'verify-signature',
        'firebase-auth',
      ]

      const mockStore = createMockRootStore({
        auth: {
          getAllSteps: jest.fn().mockReturnValue(allSteps),
          getCurrentStepInfo: jest.fn().mockReturnValue(null),
          isProgressComplete: true,
          progressError: null,
          getCompletedStepsCount: jest.fn().mockReturnValue(6),
        },
      })

      const { getByTestId } = renderWithStore(<AuthProgressIndicator />, { store: mockStore })

      await waitForMobX()

      const progressStats = getByTestId('progress-stats')
      expect(progressStats.props.children).toEqual([6, ' of ', 6, ' completed'])

      const progressComplete = getByTestId('progress-complete')
      expect(progressComplete).toBeTruthy()
    })
  })

  describe('Error Handling', () => {
    it('should handle store errors gracefully', async () => {
      const errorMessage = 'Network error occurred'
      const mockStore = createMockRootStore({
        auth: {
          getAllSteps: jest.fn().mockReturnValue(['connect-wallet', 'verify-signature']),
          getCurrentStepInfo: jest.fn().mockReturnValue(null),
          isProgressComplete: false,
          progressError: errorMessage,
          getCompletedStepsCount: jest.fn().mockReturnValue(1),
        },
      })

      const { getByTestId } = renderWithStore(<AuthProgressIndicator />, { store: mockStore })

      await waitForMobX()

      const progressError = getByTestId('progress-error')
      expect(progressError.props.children).toBe(errorMessage)

      // Should still show progress stats
      const progressStats = getByTestId('progress-stats')
      expect(progressStats).toBeTruthy()
    })

    it('should handle missing step info gracefully', async () => {
      const mockStore = createMockRootStore({
        auth: {
          getAllSteps: jest.fn().mockReturnValue(['connect-wallet', 'generate-message']),
          getCurrentStepInfo: jest.fn().mockReturnValue({
            title: undefined,
            description: undefined,
          }),
          isProgressComplete: false,
          progressError: null,
          getCompletedStepsCount: jest.fn().mockReturnValue(1),
        },
      })

      const { getByTestId } = renderWithStore(<AuthProgressIndicator />, { store: mockStore })

      await waitForMobX()

      // Should not crash, and display empty/undefined step info
      const currentStepTitle = getByTestId('current-step-title')
      expect(currentStepTitle.props.children).toBeUndefined()
    })
  })

  describe('Custom Styling', () => {
    it('should apply custom className', () => {
      const customClassName = 'p-4 bg-gray-100'
      const { getByTestId } = renderWithStore(<AuthProgressIndicator className={customClassName} />)

      const container = getByTestId('auth-progress-indicator')
      expect(container.props.className).toContain('space-y-4')
      expect(container.props.className).toContain(customClassName)
    })

    it('should render with default className when none provided', () => {
      const { getByTestId } = renderWithStore(<AuthProgressIndicator />)

      const container = getByTestId('auth-progress-indicator')
      expect(container.props.className).toBe('space-y-4 ')
    })
  })
})
