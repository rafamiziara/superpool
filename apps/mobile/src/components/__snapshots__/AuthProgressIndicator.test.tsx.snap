// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthProgressIndicator Snapshot Testing should match snapshot with active step 1`] = `
<View
  className="space-y-4 "
  testID="auth-progress-indicator"
>
  <View
    className="flex-row justify-center items-center gap-2 "
    testID="progress-indicator-container"
  >
    <View
      accessibilityLabel="Step 1 of 6 (completed)"
      className="w-2 h-2 rounded-full bg-muted-foreground/30"
      testID="progress-step-0"
    />
    <View
      accessibilityLabel="Step 2 of 6 (completed)"
      className="w-2 h-2 rounded-full bg-muted-foreground/30"
      testID="progress-step-1"
    />
    <View
      accessibilityLabel="Step 3 of 6 (current)"
      className="w-2 h-2 rounded-full bg-primary"
      testID="progress-step-2"
    />
    <View
      accessibilityLabel="Step 4 of 6 (pending)"
      className="w-2 h-2 rounded-full bg-muted-foreground/30"
      testID="progress-step-3"
    />
    <View
      accessibilityLabel="Step 5 of 6 (pending)"
      className="w-2 h-2 rounded-full bg-muted-foreground/30"
      testID="progress-step-4"
    />
    <View
      accessibilityLabel="Step 6 of 6 (pending)"
      className="w-2 h-2 rounded-full bg-muted-foreground/30"
      testID="progress-step-5"
    />
  </View>
  <View
    className="space-y-2"
  >
    <Text
      className="text-center text-sm font-medium text-foreground"
      testID="current-step-title"
    >
      Generate Auth Message
    </Text>
    <Text
      className="text-center text-xs text-muted-foreground"
      testID="current-step-description"
    >
      Creating authentication challenge
    </Text>
  </View>
  <View
    className="flex-row justify-center space-x-4"
  >
    <Text
      className="text-xs text-muted-foreground"
      testID="progress-stats"
    >
      1
       of 
      6
       completed
    </Text>
  </View>
</View>
`;

exports[`AuthProgressIndicator Snapshot Testing should match snapshot with completed state 1`] = `
<View
  className="space-y-4 "
  testID="auth-progress-indicator"
>
  <View
    className="flex-row justify-center items-center gap-2 "
    testID="progress-indicator-container"
  >
    <View
      accessibilityLabel="Step 1 of 6 (pending)"
      className="w-2 h-2 rounded-full bg-muted-foreground/30"
      testID="progress-step-0"
    />
    <View
      accessibilityLabel="Step 2 of 6 (pending)"
      className="w-2 h-2 rounded-full bg-muted-foreground/30"
      testID="progress-step-1"
    />
    <View
      accessibilityLabel="Step 3 of 6 (pending)"
      className="w-2 h-2 rounded-full bg-muted-foreground/30"
      testID="progress-step-2"
    />
    <View
      accessibilityLabel="Step 4 of 6 (pending)"
      className="w-2 h-2 rounded-full bg-muted-foreground/30"
      testID="progress-step-3"
    />
    <View
      accessibilityLabel="Step 5 of 6 (pending)"
      className="w-2 h-2 rounded-full bg-muted-foreground/30"
      testID="progress-step-4"
    />
    <View
      accessibilityLabel="Step 6 of 6 (pending)"
      className="w-2 h-2 rounded-full bg-muted-foreground/30"
      testID="progress-step-5"
    />
  </View>
  <View
    className="space-y-2"
  />
  <View
    className="flex-row justify-center space-x-4"
  >
    <Text
      className="text-xs text-muted-foreground"
      testID="progress-stats"
    >
      2
       of 
      6
       completed
    </Text>
    <Text
      className="text-xs text-green-600 font-medium"
      testID="progress-complete"
    >
      âœ“ Complete
    </Text>
  </View>
</View>
`;

exports[`AuthProgressIndicator Snapshot Testing should match snapshot with default state 1`] = `
<View
  className="space-y-4 "
  testID="auth-progress-indicator"
>
  <View
    className="flex-row justify-center items-center gap-2 "
    testID="progress-indicator-container"
  >
    <View
      accessibilityLabel="Step 1 of 6 (pending)"
      className="w-2 h-2 rounded-full bg-muted-foreground/30"
      testID="progress-step-0"
    />
    <View
      accessibilityLabel="Step 2 of 6 (pending)"
      className="w-2 h-2 rounded-full bg-muted-foreground/30"
      testID="progress-step-1"
    />
    <View
      accessibilityLabel="Step 3 of 6 (pending)"
      className="w-2 h-2 rounded-full bg-muted-foreground/30"
      testID="progress-step-2"
    />
    <View
      accessibilityLabel="Step 4 of 6 (pending)"
      className="w-2 h-2 rounded-full bg-muted-foreground/30"
      testID="progress-step-3"
    />
    <View
      accessibilityLabel="Step 5 of 6 (pending)"
      className="w-2 h-2 rounded-full bg-muted-foreground/30"
      testID="progress-step-4"
    />
    <View
      accessibilityLabel="Step 6 of 6 (pending)"
      className="w-2 h-2 rounded-full bg-muted-foreground/30"
      testID="progress-step-5"
    />
  </View>
  <View
    className="space-y-2"
  />
  <View
    className="flex-row justify-center space-x-4"
  >
    <Text
      className="text-xs text-muted-foreground"
      testID="progress-stats"
    >
      1
       of 
      6
       completed
    </Text>
  </View>
</View>
`;

exports[`AuthProgressIndicator Snapshot Testing should match snapshot with error state 1`] = `
<View
  className="space-y-4 "
  testID="auth-progress-indicator"
>
  <View
    className="flex-row justify-center items-center gap-2 "
    testID="progress-indicator-container"
  >
    <View
      accessibilityLabel="Step 1 of 6 (pending)"
      className="w-2 h-2 rounded-full bg-muted-foreground/30"
      testID="progress-step-0"
    />
    <View
      accessibilityLabel="Step 2 of 6 (pending)"
      className="w-2 h-2 rounded-full bg-muted-foreground/30"
      testID="progress-step-1"
    />
    <View
      accessibilityLabel="Step 3 of 6 (pending)"
      className="w-2 h-2 rounded-full bg-muted-foreground/30"
      testID="progress-step-2"
    />
    <View
      accessibilityLabel="Step 4 of 6 (pending)"
      className="w-2 h-2 rounded-full bg-muted-foreground/30"
      testID="progress-step-3"
    />
    <View
      accessibilityLabel="Step 5 of 6 (pending)"
      className="w-2 h-2 rounded-full bg-muted-foreground/30"
      testID="progress-step-4"
    />
    <View
      accessibilityLabel="Step 6 of 6 (pending)"
      className="w-2 h-2 rounded-full bg-muted-foreground/30"
      testID="progress-step-5"
    />
  </View>
  <View
    className="space-y-2"
  >
    <Text
      className="text-center text-xs text-destructive"
      testID="progress-error"
    >
      Authentication failed
    </Text>
  </View>
  <View
    className="flex-row justify-center space-x-4"
  >
    <Text
      className="text-xs text-muted-foreground"
      testID="progress-stats"
    >
      1
       of 
      6
       completed
    </Text>
  </View>
</View>
`;
